Defend The Divine/Assets/Scripts/Grid.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Experimental.AI;
+
+public class Grid : MonoBehaviour
+{
+    [SerializeField]
+    private Vector3 originPosition;
+    [SerializeField]
+    public int width;
+    [SerializeField]
+    public int height;
+    [SerializeField]
+    public float cellSize;
+
+    private int[,] gridArray;
+
+    private void Start()
+    {
+        gridArray = new int[width, height];
+    }
+
+    private void Update()
+    {
+        updateDebug();
+    }
+
+    /// <summary>
+    /// Returns a the lower left corner of a tile based on world position
+    /// </summary>
+    /// <param name="x"></param>
+    /// <param name="y"></param>
+    /// <returns></returns>
+    public Vector3 GetWorldPosition(int x, int y)
+    {
+        return new Vector3(x, y, 0) * cellSize + originPosition;
+    }
+
+    /// <summary>
+    /// Returns the center of a tile based on coordinates
+    /// </summary>
+    /// <param name="x"></param>
+    /// <param name="y"></param>
+    /// <returns></returns>
+    public Vector3 GetTileCenter(int x, int y)
+    {
+        return GetWorldPosition(x, y) + new Vector3(cellSize / 2f, cellSize / 2f, 0);
+    }
+
+    /// <summary>
+    /// Returns the indecies of the tile containing 'worldPosition'
+    /// </summary>
+    /// <param name="worldPosition"></param>
+    /// <param name="x"></param>
+    /// <param name="y"></param>
+    public void GetXY(Vector3 worldPosition, out int x, out int y)
+    {
+        x = Mathf.FloorToInt((worldPosition - originPosition).x / cellSize);
+        y = Mathf.FloorToInt((worldPosition - originPosition).y / cellSize);
+    }
+
+    /// <summary>
+    /// Sets the value of the grid's 'gridArray' at [x, y]
+    /// </summary>
+    /// <param name="x"></param>
+    /// <param name="y"></param>
+    /// <param name="value"></param>
+    public void SetValue(int x, int y, int value)
+    {
+        if (x >= 0 && y >= 0 && x < width && y < height)
+        {
+            gridArray[x, y] = value;
+        }
+    }
+
+    /// <summary>
+    /// Sets the value of the grid's 'gridArray' at the [x, y] tile containing
+    /// 'worldPosition'
+    /// </summary>
+    /// <param name="worldPosition"></param>
+    /// <param name="value"></param>
+    public void SetValue(Vector3 worldPosition, int value)
+    {
+        int x, y;
+        GetXY(worldPosition, out x, out y);
+        SetValue(x, y, value);
+    }
+
+    public int GetValue(int x, int y)
+    {
+        return gridArray[x, y];
+    }
+
+    public int GetValue(Vector3 worldPosition)
+    {
+        int x, y;
+        GetXY(worldPosition, out x, out y);
+        return GetValue(x, y);
+    }
+
+    private Color cellColor = Color.red;
```

Comment:
This is fine, but why is cellColor delcared here - just weird to not see it at top of file even if it's a debug